src/TMediaData.h

```diff
@@ -40,50 +40,51 @@ class TMediaData
 
     enum MediaContinue { MediaContinueDefault = true, MediaContinueRestart = false };
 
+    enum MediaFade { MediaFadeNotSet = 0 };
```

Comment:
I guess you are getting closer, but TBH I think you should abandon using an `enum` for *this* parameter altogether. Instead, since you want to have a special (well, default) initialisation value perhaps you can defined that as:
```suggestion
    static const MediaFadeNotSet = 0;
```
which can be used as `MediaFadeNotSet` within this class or `TMediaData::MediaFadeNotSet`outside...

OTOH Reading https://en.cppreference.com/w/cpp/language/enum more closely I deduce that the smallest type that can be used as the underlying type for the given value (not values, as only one is provided) is an `int` so the current code means that the ***unscoped** `enum`* `MediaFade` ***can*** take any value that an `int` can and the absolute **minimum** for that is the range -32767 to 32767 - though in all the systems we code for it is more than that!

Reply:
Made this change, thank you.